gnucash/import-export/aqb/gnc-ab-utils.c

```diff
@@ -690,6 +692,19 @@ gnc_ab_txn_to_gnc_acc(GtkWidget *parent, const AB_TRANSACTION *transaction)
     online_id = g_strconcat(bankcode ? bankcode : "",
                             accountnumber ? accountnumber : "",
                             (gchar*)NULL);
+
+    #ifdef AQBANKING6
+    gnc_acc = ieci->gnc_acc;
+    if(gnc_acc)
+    {
+      bankcode = gnc_ab_get_account_bankcode(gnc_acc);
+      accountnumber = gnc_ab_get_account_accountid(gnc_acc);
+      online_id = g_strconcat(bankcode ? bankcode : "",
+                                accountnumber ? accountnumber : "",
+                                (gchar*)NULL);
+    }
+    #endif
+
```

Comment:
@ingoha, You get the `bankcode` and `accountid` from `gnc_acc`'s KVP, construct an `online_id` from them, then search *the entire account tree* for an account with the matching KVP. Expecting to find what, if not `gnc_acc`?

Reply:
Sorry, I took some days off 8-)
@jralls Thanks for the comment - I certainly suffered from tunnel vision while coding this...

So could the algorithm (for importing a transaction) be: If gnc_acc contains bankcode and accountnumber (it does for a SWIFT transaction), use it. Else try to find the right gnc_acc using the online_id as before?

Reply:
@ingoha, You slacker! ;-)
Using the `online_id` *is* effectively using the bank code and account number. The `online_id` is just a concatenation of the two. All import methods do that; the difference is where the numbers come from.  Either way, that's not what the problem is. The problem is that the bank code and account number that AQBanking retrieves from the transactions has extra characters on it so it doesn't match the bank code and account number that AQBanking retrieves when setting up a userid so GnuCash fails to match them.

Each AQB session retrieves a bank code and account number with `AB_ImExporterAccountInfo` so we have the undecorated versions available, and we use that to find an `Account*`. The simplest fix would be to stop there and just use that as the account for the import. However that would break if the import contains transactions for a different account, so perhaps we should cache the bank id and account numbers from that and check that each transaction's bank id and account number begin with exactly those values. I proposed something link that almost 3 weeks ago and I don't think that either you or @cstim have suggested anything else that would work.

We also need to decide what to do when there's a mismatch. One option is to log an error in the tracefile and maybe raise a dialog at the end of the import telling the user that some of the transactions were dropped for having the wrong account. Another is to cycle through all of the AQB user ids, retrieving their bank codes and account numbers and doing the same begins-with match on each and if/when we find a match combine them and search for a matching account. It being possible that we wouldn't find one we'd still need to log an error and raise a warning dialog at the end.